qiskit/extensions/quantum_initializer/initializer.py

```diff
@@ -175,6 +182,13 @@ def _define_synthesis(self):
 
         return initialize_circuit
 
+    def _normalize(self, state_array):
+        """Normalizes input list or ndarray of real or complex nums.
+        Returns ndarray."""
+        norm_factor = np.linalg.norm(state_array, 2)  # l2 norm
+        normalized_state_array = np.array(state_array) / norm_factor
+        return normalized_state_array
```

Comment:
This is very minor, but if we're ensuring that `state_array` is an array at some point in this function, it's most efficient to do that first, so `np.linalg.norm` doesn't have to do the conversion internally.  We can do it safely with
```python
out = np.array(state_array)
out /= np.linalg.norm(out)
return out
```
That means that we only operate on Numpy arrays, and we don't allocate extra temporary memory by using in-place operations.

Reply:
Really great advice @jakelishman, makes sense. 

In case this is of interest to you, for the second line I'm using
`out = out / np.linalg.norm(out)`
because I got a `TypeError` using `/=`:
`TypeError: ufunc 'true_divide' output (typecode 'd') could not be coerced to provided output parameter (typecode 'l') according to the casting rule ''same_kind''`

Reply:
You must be passing an array of integers.  My line `np.array(state_array)` should become `np.array(state_array, dtype=np.complex128)` and then it'll all work.  (We actually should make sure that `state_array` _always_ becomes an array of complex, even if `normalize=False` - that might already happen in `__init__`.

The reason you got the error is related to the reason it's more efficient to use `/=`.  When you do `/=`, that's an in-place operation.  In Numpy arrays, that means that it uses the same memory.  When you do `a = a / b`, it creates a new array, with new memory.  The trick is that if you don't create a new array, then the output type needs to be the same as the input type.  `np.linalg.norm` always returns a float, but `int / float` doesn't work.  If we ensure that `state_array` is a float or complex, then it does work.  A statevector's type should be complex anyway, so it's good for us to upcast immediately.